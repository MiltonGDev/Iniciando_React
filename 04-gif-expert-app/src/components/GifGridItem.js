import React from 'react'

export const GifGridItem = ({img, tittle, url}) => {
    return (
        <div className='card animate__animated animate__fadeIn'>

            <img src={url} alt={tittle}></img>
            <p>{tittle}</p>

        </div>
    )
}
