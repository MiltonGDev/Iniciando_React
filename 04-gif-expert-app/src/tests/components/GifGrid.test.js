import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');



describe('Pruebas GifGrid', () => {
    
    const category = 'One punch';
    
    test('Haciendo match GifGrid', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category}/>)

        expect(wrapper).toMatchSnapshot();
        
    });

    test('debe de mostrar items cuando se cargan imagenes', () => {

        const gifs = [{
            id: 'ABC',
            url: 'url/xd',
            tittle: 'a tittle'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: true
        });
        
        const wrapper = shallow(<GifGrid category={category}/>)

        

    })
    
    

})



