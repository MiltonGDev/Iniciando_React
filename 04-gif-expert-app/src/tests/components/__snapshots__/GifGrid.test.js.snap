// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas GifGrid Haciendo match GifGrid 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
